vault/quotas/quotas.go

```diff
@@ -667,8 +683,20 @@ func (m *Manager) QueryResolveRoleQuotas(req *Request) (bool, error) {
 	return false, nil
 }
 
-// DeleteQuota removes a quota rule from the db for a given name
+// DeleteQuota removes a quota rule the QuotaManager's storage view and then
+// updates the associated index in memdb.
 func (m *Manager) DeleteQuota(ctx context.Context, qType string, name string) error {
+	m.quotaLock.Lock()
+	if err := m.storage.Delete(ctx, QuotaStoragePath(qType, name)); err != nil {
+		return err
+	}
+	m.quotaLock.Unlock()
```

Comment:
Now that you're performing this delete here, it seems like we probably want to remove the storage delete from `handleRateLimitQuotasDelete()` in `logical_system_quotas.go`, yes? Similar to what you did for the storage update?

Reply:
Whew, good catch! I had this in some version of this change, but I guess I dropped it somewhere. I'll push up a fix in a moment.

Reply:
Fixed in https://github.com/hashicorp/vault/pull/24381/commits/9b3abbbf8d4788ad845800f5cbd372ae39a096bc